npg_qc_viewer/lib/npg_qc_viewer/Model/LimsServer.pm

```diff
@@ -15,21 +15,10 @@ sub generate_url {
   if ($entity_type !~ /^sample$|^library$|^pool$/smx) {
     croak qq[Unknown entity type "$entity_type"];
   }
-  my $expected_type = 'npg_qc_viewer::Util::TransferObject';
-  my $actual = ref $to;
-  if ($actual ne $expected_type) {
-    croak "$expected_type is expected in second argument, got $actual";
-  }
-
-  return $to->lims_live ? $self->_link_sscape($entity_type, $to) : q[];
-}
-
-sub _link_sscape {
-  my ($self, $entity_type, $to) = @_;
 
   my $url = npg_qc_viewer->config->{'Model::LimsServer'}->{'sscape_url'};
   my $link = q[];
-  if ($url) {
+  if ($url && $entity_type) {
```

Comment:
I see your point, but `!~` above would trigger on `undef` I would expect, probably with a warning.

Reply:
`$entity_type ||= q[];` prevents this
By the time the code reaches line 21, we do not really need to check the value of `$entity_type`